### DISCUSSION TITLE
Algorand Mainnet Not Found/Supported

### DISCUSSION THREAD

**Discussion by Dilosen** [WalletConnect/WalletConnectUnity#222] [Q&A]
To recreate the issue: 

Install the package, download the walletconnect sample and run in either PC or Android build modes. 

Wallets tested: Ledger, Pera, Metamask

When using PC build version:

- QR code is used to connect to the wallet.
- error message in wallet: "there is a mismatch. Either the dapp is on testnet or the wallet is on mainnet or visa versa. "

When using Android build (either in editor or on device):

- no QR, only able to click on wallet from the list or copy the link
- error message in wallet : "there is a mismatch. Either the dapp is on testnet or the wallet is on mainnet or visa versa. "

The sample includes the Algorand Mainnet, which can be seen in the script.

Both Wallets and the Sample are all running on the mainnet out of the box, so i don't understand why there is a mismatch because no where do we disclose the testnet. Could there be a mismatch with the blockchain address?

algorand:wGHE2Pwdvd7S12BL5FaOP20EGYesN73k - this is the mainet one provided by WalletConnect [here](https://docs.walletconnect.com/cloud/chains/chain-list)

Not sure exactly if this address is being called because the script example has: 
                    
                  `  new BlockExplorer("Pera Explorer", "https://explorer.perawallet.app/"),
                    "https://mainnet-api.algonode.cloud",`

**Comment by skibitsky** [comment]
Hello,

MetaMask Mobile is an EVM wallet; it does not support Algorand.

The connection error in Ledger Live lists all chains their WalletConnection integration supports, Algorand isn't on the list.

<img width=500 src=https://github.com/WalletConnect/WalletConnectUnity/assets/15127974/10e67385-b474-46a4-a30b-e9d4b244d945 />

As for the Pera Wallet, I believe that it used to work. The network ID used by the sample is correct. It's what Algorand used in their [CAIP-2 document](https://github.com/ChainAgnostic/namespaces/blob/main/algorand/caip2.md#test-cases). Maybe reach out to them and ask what chain ID and methods their wallet supports?

**Comment by Dilosen** [comment]
Hi There, 

" their WalletConnection integration supports," ah this makes sense. I was confused because they have Algorand wallets and I can send/recieve on that network. This clarifies that part. 

To be honest, at this stage it seems more work that it should be just to get it running. I'm going to explore other options that are available and intergrated. 

Thanks!
